smart_contract/executor/src/default.rs

```diff
@@ -295,22 +439,34 @@ pub mod account {
     ) {
         let account_id = isi.object_id();
 
-        if is_genesis(executor) {
-            execute!(executor, isi);
-        }
-        match is_account_owner(account_id, authority) {
-            Err(err) => deny!(executor, err),
-            Ok(true) => execute!(executor, isi),
-            Ok(false) => {}
-        }
-        let can_unregister_user_account = tokens::account::CanUnregisterAccount {
-            account_id: account_id.clone(),
-        };
-        if can_unregister_user_account.is_owned_by(authority) {
-            execute!(executor, isi);
+        if is_genesis(executor)
+            || match is_account_owner(account_id, authority) {
+                Err(err) => deny!(executor, err),
+                Ok(is_account_owner) => is_account_owner,
+            }
+            || {
+                let can_unregister_user_account = tokens::account::CanUnregisterAccount {
+                    account_id: account_id.clone(),
+                };
+                can_unregister_user_account.is_owned_by(authority)
+            }
+        {
+            if executor.verdict().is_ok() {
+                if let Err(err) = isi.execute() {
+                    executor.deny(err);
+                }
+            }
```

Comment:
I would prefer removing a token after it has been revoked from all the entities. Just move this piece of code after the `for loop`. Also in other cases